src/FileExportManager.js

```diff
@@ -148,115 +150,124 @@ class FileExportManager {
-                    this.emit('update', ProgressMessages.ExportSession(finishedSessions.length + 1, sessionExportTotal));
-                    finishedSessions.push(prefix);
-                  }
-
-                  try {
-                    // gather one promise for each exported file
-                    return exportFile(
+        
+                const exportFormats = [
+                  ...(this.exportOptions.exportGraphML ? ['graphml'] : []),
+                  ...(this.exportOptions.exportCSV ? ['ego'] : []),
+                  ...(this.exportOptions.exportCSV.adjacencyMatrix ? ['adjacencyMatrix'] : []),
+                  ...(this.exportOptions.exportCSV.attributeList ? ['attributeList'] : []),
+                  ...(this.exportOptions.exportCSV.edgeList ? ['edgeList'] : []),
+                ];
+        
+                // Returns promise resolving to filePath for each format exported
+                // ['file1', ['file1_person', 'file1_place']]
+                exportFormats.forEach((format) => {
+                  // partitioning network based on node and edge type so we can create
+                  // an individual export file for each type
+                  const partitionedNetworks = partitionNetworkByType(
+                    protocol.codebook,
+                    session,
+                    format,
+                  );
+        
+                  partitionedNetworks.forEach((partitionedNetwork) => {
+                    const partitionedEntity = partitionedNetwork.partitionEntity;
+                    promisedExports.push(() => exportFile(
                       prefix,
                       partitionedEntity,
                       format,
                       tmpDir,
                       partitionedNetwork,
                       protocol.codebook,
                       this.exportOptions,
-                    );
-                  } catch (error) {
-                    this.emit('error', `Encoding ${prefix} failed: ${error.message}`);
-                    return Promise.reject(error);
-                  }
+                    ).then(result => {
+                      if (!finishedSessions.includes(prefix)) {
+                        this.emit('session-exported', session.sessionVariables.sessionId);
+                        this.emit('update', ProgressMessages.ExportSession(finishedSessions.length + 1, sessionExportTotal));
+                        finishedSessions.push(prefix);
+                      }
+                      return result;
+                    }).catch((error) => {
+                      this.emit('error', `Encoding ${prefix} failed: ${error.message}`);
+                      return Promise.reject(error);
+                    }));
+                  });
                 });
               });
-            });
-          }),
-        );
-
-        return Promise.allSettled(promisedExports);
+          })
+          
+          q.push(promisedExports);
+          q.drain().then(() => {
```

Comment:
It is possible to to `await q.drain()`, but it would require the parent promise to be re written as `.then(async() => {...` which I read elsewhere is not a good pattern. Open to suggestions.